test/helpers/graphql_helper.rb

```diff
@@ -3,7 +3,7 @@
 RSpec::Matchers.define :have_graphql_span do
   match do |actual|
     Span.all.any? do |span|
-      span.name == "graphql.execute" && span.attributes["graphql.source"] == actual
+      span.name == "graphql.parse" && span.attributes["graphql.source"] == actual
```

Comment:
Could you explain this change? Did the update change which span has the GraphQL query?

It should be fine, because [the agent doesn't actually care](https://github.com/appsignal/appsignal-agent/blob/main/common/src/opentelemetry/extractors/nodejs/graphql.rs#L25) which event has `graphql.source` in it, as long as one of them does.

But it seems strange to me that the `graphql.execute` event no longer has it -- we haven't updated the GraphQL instrumentation here, just Yoga.

Reply:
This is the span data I get back in this test app:

```ruby
<Span:0x0000aaaac4f35088
  @name="middleware - query",
  @id="b05c41d1c473e236",
  @parent_id="8618157b5c2b6972",
  @attributes={
  "http.route"=>"/",
  "express.name"=>"query",
  "express.type"=>"middleware"
  },
  @events=[],
  @instrumentation_library_name="@opentelemetry/instrumentation-express">,
  #<Span:0x0000aaaac4f34ae8
  @name="middleware - expressInit",
  @id="b8422b98e2f3c13d",
  @parent_id="8618157b5c2b6972",
  @attributes={"http.route"=>"/",
               "express.name"=>"expressInit",
               "express.type"=>"middleware"
  },
  @events=[],
  @instrumentation_library_name="@opentelemetry/instrumentation-express"
  >,
<Span:0x0000aaaac4f345e8
  @name="graphql.parse",
  @id="571e6bd03139e559",
  @parent_id="8618157b5c2b6972",
  @attributes={"graphql.source"=>"{ books { title author } }"},
  @events=[],
  @instrumentation_library_name="@opentelemetry/instrumentation-graphql"
>,
<Span:0x0000aaaac4f34138
  @name="graphql.validate",
  @id="137b9aeace290dd8",
  @parent_id="8618157b5c2b6972",
  @attributes={},
  @events=[],
  @instrumentation_library_name="@opentelemetry/instrumentation-graphql"
>,
<Span:0x0000aaaac4f3fb78
  @name="middleware - genericRequestHandler",
  @id="758eca5ecae952bb",
  @parent_id="8618157b5c2b6972",
  @attributes={"http.route"=>"/graphql",
                "express.name"=>"genericRequestHandler",
                "express.type"=>"middleware"
  },
  @events=[],
  @instrumentation_library_name="@opentelemetry/instrumentation-express"
>,
<Span:0x0000aaaac4f3f178 @name="POST /graphql",
  @id="8618157b5c2b6972",
  @parent_id=nil,
  @attributes={"http.url"=>"http://app:4001/graphql",
                "http.host"=>"app:4001",
                "net.host.name"=>"app",
                "http.method"=>"POST",
                "http.scheme"=>"http",
                "http.target"=>"/graphql",
                "http.user_agent"=>"http.rb/5.1.0",
                "http.request_content_length_uncompressed"=>38,
                "http.flavor"=>"1.1",
                "net.transport"=>"ip_tcp",
                "http.request.header.connection"=>["close"],
                "http.request.header.content_length"=>["38"],
                "net.host.ip"=>"::ffff:172.20.0.2",
                "net.host.port"=>4001,
                "net.peer.ip"=>"::ffff:172.20.0.3",
                "net.peer.port"=>59906,
                "http.status_code"=>200,
                "http.status_text"=>"OK",
                "http.route"=>"/graphql"
  },
  @events=[],
  @instrumentation_library_name="@opentelemetry/instrumentation-http"
>
```
There is no `graphql.execute` span here.

Also seems to be present for other apps otherwise more test apps would have failed.

Reply:
Huh. I wonder why that changed. Regardless, it doesn't seem concerning. 🤷